src/field_theory/intermediate_field.lean

```diff
@@ -305,6 +305,10 @@ S.to_subalgebra.val
 
 @[simp] lemma val_mk {x : L} (hx : x ∈ S) : S.val ⟨x, hx⟩ = x := rfl
 
+lemma val_inj {K L : Type*} [field K] [field L] [algebra K L]
```

Comment:
Do we have this lemma for `subalgebta`? I think usually we don't bother with this type of lemma; instead of saying `foo_hom` is injective we just make do with saying `foo` is injective, which is `subtype.coe_injective`